utils.py

```diff
@@ -88,8 +89,12 @@ def compute_ram_memory_stats(byte_scaler =1024 ** 3):
 
 
 def compute_memory_fragmentation():
-	snapshot = torch.cuda.memory_snapshot()
-	return sum(b['allocated_size'] for b in snapshot) / sum(b['total_size'] for b in snapshot)
+    lib = ctypes.cdll.LoadLibrary(None)
+    free_mem = ctypes.c_long()
+    total_mem = ctypes.c_long()
+    lib.cudaMemGetInfo(ctypes.byref(free_mem), ctypes.byref(total_mem))
+    return (total_mem.value - free_mem.value) / (1024 ** 3)
```

Comment:
either return bytes or move `1024 ** 3` to a `scaler` argument, like we have for `compute_model_capacity`

Reply:
function semantics is changed